core/lib/multivm/src/tracers/call_tracer/mod.rs

```diff
@@ -18,11 +23,20 @@ struct FarcallAndNearCallCount {
     near_calls_after: usize,
 }
 
+impl Drop for CallTracer {
+    fn drop(&mut self) {
+        CALL_METRICS.call_stack_depth.observe(self.max_stack_depth);
+        CALL_METRICS.max_near_calls.observe(self.max_near_calls);
+    }
+}
```

Comment:
Stupid question: Is it correct that a single tracer is always used for exactly one transaction? Otherwise, we'd get bogus zero values for metrics and/or "glue" some values together.

Reply:
As far as I understood - yes, but maybe I am wrong